TargetSDK 34 - FOREGROUND_SERVICE_MEDIA_PLAYBACK permission required

### Describe your problem!

c:geo crashes when I hit start talking, whether from the main cache page or the compass screen

### How to reproduce?

1. Open any cache 
2. Open menu
3. Select 'start talking'
Alternatively 

1. Open any cache 
2. Open navigation menu
3. Select 'Compass'
4. Click on start talking icon

### Actual result after these steps?

![Screenshot_20240724-070142](https://github.com/user-attachments/assets/67c4706c-c4f0-4e73-89ff-f3e5869e67e6)


### Expected result after these steps?

The app ought start talking

### Reproducible

Yes

### c:geo Version

2024.07.20-RC

### System information

```text
## System information

c:geo version: 2024.07.20-RC

Device:
-------
- Device type: Pixel 6 Pro (raven, google)
- Available processors: 8
- Android version: 14
- Android build: UQ1A.240205.002
- Screen resolution: 1440x2807px (411x802dp)
- Pixel density: 3.5
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: enabled - 24.26.32 (190400-650348549)
- Memory:  Available:3.6 GB, Total:11.3 GB, Threshold: 216.0 MB, low:false

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:277
- Set language: en
- System date format: M/d/yy
- Time zone: CDT
- Debug mode active: no
- Log Settings: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
- Last manual backup: Feb 12, 2023, 4:53 PM
- Last auto backup: Jul 19, 2:59 PM
- Routing mode: Straight (shortest.brf / trekking.brf / car-eco.brf / - / -)
- Live map mode: true
- Use unified map: true
- OSM multi-threading: true / threads: 4
- Map: Google: Satellite
  - Id: cgeo.geocaching.maps.google.v2.GoogleMapProvider$GoogleSatelliteSource
  - Atts: none
  - Theme: none

Filters:
-------
- Hide waypoints: visited
- LIVE: None ({"advanced":true,"inconclusive":false})
- OFFLINE: None ({"advanced":true,"inconclusive":false})
- Additional stored filters: 9

Services:
-------
- Geocaching sites enabled:
   - geocaching.com: Logged in (Login OK) / PREMIUM
- Geocaching.com date format: dd MMM yy
- Geocaching.com website language: en-US
- Last login error on geocaching.com: communication error (2024-06-25 23-10)
- Last successful login on geocaching.com: 2024-07-24 07-02
- Routing: internal / BRouter installed: true

Permissions
-------
- ACCESS_COARSE_LOCATION: granted
- ACCESS_FINE_LOCATION: granted
- READ_CONTACTS: DENIED
- WRITE_EXTERNAL_STORAGE: DENIED
- POST_NOTIFICATIONS: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (3.4 GB free) v3 internal isDir(10 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (3.4 GB free) v3 internal isDir(10 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (3.4 GB free) v3 external non-removable isDir(1172 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (3.4 GB free) v3 internal isDir(0 entries)
- Map render theme path: 
- Public Folders: #13
  - BASE: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo (av:true, files:>=13, size:>=171.9 MB, free:>=3.4 GB)
  - OFFLINE_MAPS: content://com.android.externalstorage.documents/tree/primary:Locus/mapsVector/document/primary:Locus/mapsVector (av:false, files:>=0, size:>=0 B, free:>=-1 B)
  - OFFLINE_MAP_THEMES: file:///data/user/0/cgeo.geocaching/files/public/OFFLINE_MAP_THEMES (av:true, files:>=0, size:>=0 B, free:>=3.4 GB)
  - OFFLINE_MAP_SHADING: file:///data/user/0/cgeo.geocaching/files/public/OFFLINE_MAP_SHADING (av:true, files:>=0, size:>=0 B, free:>=3.4 GB)
  - LOGFILES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/logfiles (av:true, files:>=0, size:>=0 B, free:>=3.4 GB)
  - GPX: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/gpx (av:true, files:>=5, size:>=540.7 KB, free:>=3.4 GB)
  - BACKUP: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/backup (av:true, files:>=4, size:>=58.6 MB, free:>=3.4 GB)
  - FIELD_NOTES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/field-notes (av:true, files:>=0, size:>=0 B, free:>=3.4 GB)
  - SPOILER_IMAGES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/GeocachePhotos (av:true, files:>=0, size:>=0 B, free:>=3.4 GB)
  - ROUTING_BASE: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/routing (av:true, files:>=11, size:>=136.1 MB, free:>=3.4 GB)
  - ROUTING_TILES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/routing/segments4 (av:true, files:>=3, size:>=136.1 MB, free:>=3.4 GB)
  - WHERIGO: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/wherigo (av:true, files:>=0, size:>=0 B, free:>=3.4 GB)
  - TEST_FOLDER: file:///data/user/0/cgeo.geocaching/files/unittest (av:true, files:>=0, size:>=0 B, free:>=3.4 GB)
- PersistedDocumentUris: #2
  - PROXIMITY_NOTIFICATION_FAR: null
  - PROXIMITY_NOTIFICATION_CLOSE: null
- Persisted Uri Permissions: #1
  - content://com.android.externalstorage.documents/tree/primary:cgeo (Feb 18, 2023, 8:15 PM):RW

Database
-------
- File: /data/user/0/cgeo.geocaching/databases/data (v103, Size:35.5 MB) on system internal storage
- Data: {cg_attributes=6777, cg_caches=1487, cg_caches_lists=1727, cg_categories=0, cg_extension=38, cg_filters=9, cg_lists=77, cg_logCount=6526, cg_logImages=12898, cg_logs=45244, cg_logs_offline=2, cg_logs_offline_images=0, cg_logs_offline_trackables=6, cg_route=0, cg_search_destination_history=0, cg_spoilers=1544, cg_trackables=1511, cg_trail_history=700, cg_variables=70, cg_waypoints=704}
- Extension Data: {DBEXTENSION_DOWNGRADEABLE_DBVERSION=1, DBEXTENSION_EMOJILRU=10, DBEXTENSION_FOUNDNUM=1, DBEXTENSION_LAST_TRACKABLE_ACTION=12, DBEXTENSION_ONE_TIME_DIALOGS=8, DBEXTENSION_POCKETQUERY_HISTORY=6}

--- End of system information ---
```


### Additional Information

_No response_

Sorry for the delay

On Wed, Jul 24, 2024, 11:07 AM Lineflyer ***@***.***> wrote:

> Can you please send us a log either via mail to ***@***.*** with
> reference to this issue?
>
> How To:
>
>    - Repeat the problematic action
>    - Go to c:geo Menu - Settings - System and select "Create Logfile"
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cgeo/cgeo/issues/15900#issuecomment-2248398782>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABLRJXVWR3GUTGOLRWY4YGDZN7GNRAVCNFSM6AAAAABLMK2WQKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENBYGM4TQNZYGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

--------- beginning of crash
07-24 06:34:50.724 19614 19614 E AndroidRuntime: FATAL EXCEPTION: main
07-24 06:34:50.724 19614 19614 E AndroidRuntime: Process: cgeo.geocaching, PID: 19614
07-24 06:34:50.724 19614 19614 E AndroidRuntime: java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{8cc12ef 19614:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{8cc12ef 19614:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:775)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	... 9 more
07-24 06:34:50.724 19614 19614 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2669)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2359)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1687)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13362)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3400)
07-24 06:34:50.724 19614 19614 E AndroidRuntime: 
07-24 06:34:51.115 19961 19961 E AndroidRuntime: FATAL EXCEPTION: main
07-24 06:34:51.115 19961 19961 E AndroidRuntime: Process: cgeo.geocaching, PID: 19961
07-24 06:34:51.115 19961 19961 E AndroidRuntime: java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{b953048 19961:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{b953048 19961:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:775)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 06:34:51.115 19961 19961 E AndroidRuntime: 	... 9 more
07-24 06:52:06.396 21731 21731 E AndroidRuntime: FATAL EXCEPTION: main
07-24 06:52:06.396 21731 21731 E AndroidRuntime: Process: cgeo.geocaching, PID: 21731
07-24 06:52:06.396 21731 21731 E AndroidRuntime: java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{67db97c 21731:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{67db97c 21731:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:775)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	... 9 more
07-24 06:52:06.396 21731 21731 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2669)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2359)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1687)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13362)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3400)
07-24 06:52:06.396 21731 21731 E AndroidRuntime: 
07-24 06:52:06.911 21927 21927 E AndroidRuntime: FATAL EXCEPTION: main
07-24 06:52:06.911 21927 21927 E AndroidRuntime: Process: cgeo.geocaching, PID: 21927
07-24 06:52:06.911 21927 21927 E AndroidRuntime: java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{17b1b34 21927:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{17b1b34 21927:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:775)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 06:52:06.911 21927 21927 E AndroidRuntime: 	... 9 more
07-24 07:01:34.435 21990 21990 E AndroidRuntime: FATAL EXCEPTION: main
07-24 07:01:34.435 21990 21990 E AndroidRuntime: Process: cgeo.geocaching, PID: 21990
07-24 07:01:34.435 21990 21990 E AndroidRuntime: java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{93aeb36 21990:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{93aeb36 21990:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:775)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	... 9 more
07-24 07:01:34.435 21990 21990 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2669)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2359)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1687)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13362)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3400)
07-24 07:01:34.435 21990 21990 E AndroidRuntime: 
07-24 07:01:34.931 24307 24307 E AndroidRuntime: FATAL EXCEPTION: main
07-24 07:01:34.931 24307 24307 E AndroidRuntime: Process: cgeo.geocaching, PID: 24307
07-24 07:01:34.931 24307 24307 E AndroidRuntime: java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{8400811 24307:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{8400811 24307:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:775)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 07:01:34.931 24307 24307 E AndroidRuntime: 	... 9 more
--------- beginning of main
07-24 11:51:25.498 30081 30081 W cgeo    : [main] SpeechService - WakeLock acquired
07-24 11:51:25.508 30081 30081 E AndroidRuntime: FATAL EXCEPTION: main
07-24 11:51:25.508 30081 30081 E AndroidRuntime: Process: cgeo.geocaching, PID: 30081
07-24 11:51:25.508 30081 30081 E AndroidRuntime: java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{959ac87 30081:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{959ac87 30081:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:775)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	... 9 more
07-24 11:51:25.508 30081 30081 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2669)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2359)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1687)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13362)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3400)
07-24 11:51:25.508 30081 30081 E AndroidRuntime: 
07-24 11:51:25.508 30081 30081 W cgeo    : [main] UncaughtException
07-24 11:51:25.508 30081 30081 W cgeo    : java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{959ac87 30081:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.os.Looper.loop(Looper.java:294)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at java.lang.reflect.Method.invoke(Native Method)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 11:51:25.508 30081 30081 W cgeo    : Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{959ac87 30081:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.app.Service.startForeground(Service.java:775)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 11:51:25.508 30081 30081 W cgeo    : 	... 9 more
07-24 11:51:25.508 30081 30081 W cgeo    : Caused by: android.os.RemoteException: Remote stack trace:
07-24 11:51:25.508 30081 30081 W cgeo    : 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2669)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2359)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1687)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13362)
07-24 11:51:25.508 30081 30081 W cgeo    : 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3400)
07-24 11:51:25.508 30081 30081 W cgeo    : 
07-24 11:51:25.836 27683 27683 I cgeo    : [main] [CtxLog]ContentStorage.init:START
07-24 11:51:25.839 27683 27683 I cgeo    : [main] [CtxLog]ContentStorage.init:END (3ms)
07-24 11:51:25.850 27683 27683 I cgeo    : [Log] Logging set: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
07-24 11:51:26.302 27683 27683 I cgeo    : [Log] Logging set: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
07-24 11:51:26.302 27683 27683 I cgeo    : [Log] No logging config file 'log-properties.txt' found at /cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles], using defaults
07-24 11:51:26.302 27683 27683 I cgeo    : [main] ---------------- CGeoApplication: startup -------------
07-24 11:51:26.302 27683 27683 E cgeo    : [main] c:geo version 2024.07.20-RC
07-24 11:51:26.302 27683 27683 I cgeo    : [main] [CtxLog]CGeoApplication.onCreate:START
07-24 11:51:26.305 27683 27683 I cgeo    : [main] [CtxLog]DataStore.init:START
07-24 11:51:26.310 27683 27683 I cgeo    : [main] [DB] Current Database Version: 103
07-24 11:51:26.317 27683 27683 I cgeo    : [main] [CtxLog]DataStore.init:END (12ms)
07-24 11:51:26.332 27683 27683 I cgeo    : [main] LooperLogger: started for Thread: main
07-24 11:51:26.332 27683 27683 I cgeo    : [main] [CtxLog]CGeoApplication.onCreate:END (30ms)
07-24 11:51:26.618 27683 27683 W cgeo    : [main] ContentStorage[suppressedForUser]: Folder 'OFFLINE_MAPS: /Locus/mapsVector (User-Defined)[/Locus/mapsVector[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector::]](content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector/document/primary%3ALocus%2FmapsVector)' could not be accessed
07-24 11:51:26.629 27683 27683 I cgeo    : [main] [CtxLog]DataStore.cleanIfNeeded: cleans DB:START
07-24 11:51:26.629 27683 27683 I cgeo    : [main] [CtxLog]DataStore.cleanIfNeeded: cleans DB:END (0ms)
07-24 11:51:26.632 27683 27683 I cgeo    : [main] LocalStorage: current Version: 3, expected Version: 3
07-24 11:51:26.665 27683 27683 W cgeo    : [main] SpeechService - WakeLock acquired
07-24 11:51:26.675 27683 27683 E AndroidRuntime: FATAL EXCEPTION: main
07-24 11:51:26.675 27683 27683 E AndroidRuntime: Process: cgeo.geocaching, PID: 27683
07-24 11:51:26.675 27683 27683 E AndroidRuntime: java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{55f247f 27683:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:294)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{55f247f 27683:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:775)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	... 9 more
07-24 11:51:26.675 27683 27683 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2669)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2359)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1687)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13362)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3400)
07-24 11:51:26.675 27683 27683 E AndroidRuntime: 
07-24 11:51:26.675 27683 27683 W cgeo    : [main] UncaughtException
07-24 11:51:26.675 27683 27683 W cgeo    : java.lang.RuntimeException: Unable to create service cgeo.geocaching.speech.SpeechService: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{55f247f 27683:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4705)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2289)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.os.Looper.loopOnce(Looper.java:205)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.os.Looper.loop(Looper.java:294)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.app.ActivityThread.main(ActivityThread.java:8248)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at java.lang.reflect.Method.invoke(Native Method)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
07-24 11:51:26.675 27683 27683 W cgeo    : Caused by: java.lang.SecurityException: Starting FGS with type mediaPlayback callerApp=ProcessRecord{55f247f 27683:cgeo.geocaching/u0a277} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK] 
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.os.Parcel.createException(Parcel.java:3041)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.os.Parcel.readException(Parcel.java:3024)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.os.Parcel.readException(Parcel.java:2966)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6790)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.app.Service.startForeground(Service.java:775)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at cgeo.geocaching.speech.SpeechService.onCreate(SpeechService.java:127)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4692)
07-24 11:51:26.675 27683 27683 W cgeo    : 	... 9 more
07-24 11:51:26.675 27683 27683 W cgeo    : Caused by: android.os.RemoteException: Remote stack trace:
07-24 11:51:26.675 27683 27683 W cgeo    : 	at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2669)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2359)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1687)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13362)
07-24 11:51:26.675 27683 27683 W cgeo    : 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3400)
07-24 11:51:26.675 27683 27683 W cgeo    : 
07-24 11:51:41.127 27994 27994 I cgeo    : [main] [CtxLog]ContentStorage.init:START
07-24 11:51:41.129 27994 27994 I cgeo    : [main] [CtxLog]ContentStorage.init:END (2ms)
07-24 11:51:41.133 27994 27994 I cgeo    : [Log] Logging set: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
07-24 11:51:41.222 27994 27994 I cgeo    : [Log] Logging set: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
07-24 11:51:41.222 27994 27994 I cgeo    : [Log] No logging config file 'log-properties.txt' found at /cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles], using defaults
07-24 11:51:41.222 27994 27994 I cgeo    : [main] ---------------- CGeoApplication: startup -------------
07-24 11:51:41.222 27994 27994 E cgeo    : [main] c:geo version 2024.07.20-RC
07-24 11:51:41.223 27994 27994 I cgeo    : [main] [CtxLog]CGeoApplication.onCreate:START
07-24 11:51:41.225 27994 27994 I cgeo    : [main] [CtxLog]DataStore.init:START
07-24 11:51:41.229 27994 27994 I cgeo    : [main] [DB] Current Database Version: 103
07-24 11:51:41.229 27994 27994 I cgeo    : [main] [CtxLog]DataStore.init:END (4ms)
07-24 11:51:41.232 27994 27994 I cgeo    : [main] LooperLogger: started for Thread: main
07-24 11:51:41.232 27994 27994 I cgeo    : [main] [CtxLog]CGeoApplication.onCreate:END (9ms)
07-24 11:51:41.498 27994 27994 W cgeo    : [main] ContentStorage[suppressedForUser]: Folder 'OFFLINE_MAPS: /Locus/mapsVector (User-Defined)[/Locus/mapsVector[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector::]](content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector/document/primary%3ALocus%2FmapsVector)' could not be accessed
07-24 11:51:41.504 27994 27994 I cgeo    : [main] [CtxLog]DataStore.cleanIfNeeded: cleans DB:START
07-24 11:51:41.504 27994 27994 I cgeo    : [main] [CtxLog]DataStore.cleanIfNeeded: cleans DB:END (0ms)
07-24 11:51:41.506 27994 27994 I cgeo    : [main] LocalStorage: current Version: 3, expected Version: 3
07-24 11:51:52.660 27994 27994 W cgeo    : [main] ContentStorage[suppressedForUser]: Folder 'OFFLINE_MAPS: /Locus/mapsVector (User-Defined)[/Locus/mapsVector[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector::]](content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector/document/primary%3ALocus%2FmapsVector)' could not be accessed
07-24 11:51:52.668 27994 27994 I cgeo    : [main] [CtxLog]AbstractLocusApp.init (cgeo.geocaching.apps.navi.LocusApp):START
07-24 11:51:52.737 27994 27994 I cgeo    : [main] [CtxLog]AbstractLocusApp.init (cgeo.geocaching.apps.navi.LocusApp):END (69ms)V:menion.android.locus.pro: 3.70.9(69ms);
07-24 11:51:54.092 27994 27994 E cgeo    : [main] new SeekbarUI
07-24 11:51:54.112 27994 27994 E cgeo    : [main] new SeekbarUI
07-24 11:52:31.783 27994 27994 I cgeo    : [Log] Logging set: debug=true, minLevel=DEBUG, minAddCallerInfo=DEBUG, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=true
07-24 11:52:33.487 27994 27994 D cgeo    : [main] LooperLogger: total:#8000/3691ms/avg 0ms {LooperLogger.process:78/LooperLogger.$r8$lambda$IQJlhstRB0ylWGjFWsOcbpEUIlc:0/LooperLogger$$ExternalSyntheticLambda0.println:0/Looper.loopOnce:240/Looper.loop:294/ActivityThread.main:8248/Method.invoke:-2/RuntimeInit$MethodAndArgsCaller.run:552}
07-24 11:52:38.657 27994 28037 I cgeo    : [RxCachedThreadScheduler-1] [LogCat]Issuing command: [logcat, -d, AndroidRuntime:E, cgeo:D, cgeo.geocachin:I, *:S, -f, /data/user/0/cgeo.geocaching/cache/cgeo_tempfile_1550922491502231766.tmp] {DebugUtils.lambda$createLogcatHelper$4:100/DebugUtils.$r8$lambda$k2i1lIY1xmaeW1vUjc05PtMUJJY:0/DebugUtils$$ExternalSyntheticLambda1.run:0/AndroidRxUtils.lambda$andThenOnUi$2:93/AndroidRxUtils.$r8$lambda$bqzWo9nlvU86BvboB-9jhDWp5vc:0/AndroidRxUtils$$ExternalSyntheticLambda0.run:0/ScheduledRunnable.run:65/ScheduledRunnable.call:56}

--- a/main/AndroidManifest.xml
+++ b/main/AndroidManifest.xml
@@ -10,11 +10,13 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.WAKE_LOCK" />
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC" android:minSdkVersion="34" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_MEDIA_PLAYBACK" android:minSdkVersion="34" />
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_LOCATION" android:minSdkVersion="34" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.NFC" />
-    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_DATA_SYNC" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:maxSdkVersion="28" />
     <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
      <uses-permission android:name="android.permission.READ_CONTACTS" />

